### PR TITLE
Trigger 'CommandInvoker.OnPreTruncateCommandTable' event before truncating the command table

### PR DESCRIPTION
**Pull request by chewong**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5460
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 1]
```diff
@@ -194,11 +194,15 @@ def execute(self, args):
         import knack.events as events
         from knack.util import CommandResultItem, todict
 
+        events.EVENT_INVOKER_PRE_TRUNCATE_CMD_TBL = 'CommandInvoker.OnPreTruncateCommandTable'
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: Do we have a centralized place to put the event definition? This shouldn't be littered around.

Comment (chewong) [reply src/azure-cli-core/azure/cli/core/commands/__init__.py]: I don't think there is a centralized place for knack event definitions in the core, they are defined in the [knack repo](https://github.com/Microsoft/knack/blob/master/knack/events.py). I thought we were planning to put the new event in the core first before moving it to knack

Comment (derekbekoe) [reply src/azure-cli-core/azure/cli/core/commands/__init__.py]: @cheeong Yes you're correct but there's no need to add a new property to `knack.events....`.
You should be able to just define your own string and raise that event directly.

When/if we do move it to knack, then you can use it from there.
Does that make sense?

Comment (chewong) [reply src/azure-cli-core/azure/cli/core/commands/__init__.py]: I see, thanks!

**Comment by troydai** [comment]
@chewong please paste the link to the knack's PR in the comments so that we can record the associations.

**Comment by chewong** [comment]
Created a PR on knack: https://github.com/Microsoft/knack/pull/72

**Comment by tjprescott** [comment]
Knack doesn't do command table truncation so this event is not relevant to Knack.

**Pull request merged**